submissions/Eugene-CG/OOJS-Frogger-Game/app.js

```diff
@@ -0,0 +1,110 @@
+const charactersSize = {
+  charboy: 50,
+  bug: 35,
+  sizeDiff: 15,
+};
+const tileSize = {
+  width: 101,
+  height: 85,
+};
+const startPlayerPosition = {
+  x: tileSize.width * 2,
+  y: tileSize.height * 4 - charactersSize.charboy,
+};
+const gameBoard = {
+  left: 0,
+  top: 0,
+  right: tileSize.width * 5,
+  bottom: tileSize.height * 6,
+};
+class Character {
+  constructor(x, y, sprite) {
+    this.x = x;
+    this.y = y;
+    this.sprite = sprite;
+  }
+  render() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+  }
+  update() {}
+}
+class Enemy extends Character {
+  constructor(x, y, speed) {
+    super(x, y, "images/enemy-bug.png");
+    this.speed = speed;
+  }
+  checkCollision() {
+    if (
+      this.x +
+        tileSize.width -
+        charactersSize.charboy +
+        charactersSize.sizeDiff >
+        player.x &&
+      this.x - charactersSize.charboy - charactersSize.sizeDiff < player.x &&
+      this.y === player.y + charactersSize.sizeDiff
+    ) {
+      setTimeout(() => {
+        player.x = tileSize.width * 2;
+        player.y = tileSize.height * 4 - charactersSize.charboy;
+      }, 100);
+    }
+  }
+  update(dt) {
+    this.x += this.speed * dt;
+    if (this.x > tileSize.width * 5) this.x = -tileSize.width;
+    this.checkCollision();
+  }
+}
+class Player extends Character {
+  constructor(x, y) {
+    super(x, y, "images/char-boy.png");
+  }
+  reset() {
```

Comment:
Fix method name)

Reply:
> 

Changed from "reset" to "resetToStartPosition"

Reply:
:3